platforms/software/dependency-management/src/main/java/org/gradle/api/internal/catalog/DelegatingProjectDependency.java

```diff
@@ -196,16 +190,22 @@ public void because(@Nullable String reason) {
     }
 
     @Override
+    @Deprecated
+    @SuppressWarnings("deprecation")
```

Comment:
💅 Suppressing the deprecation warning isn't necessary if these methods themselves are deprecated.  Applies throughout